### PR TITLE
CEP-492/496 updates [0.17.12]

### PR DESCRIPTION
**Pull request by StoyanTc**
- Added CEP-492/496 changes.
- Added some missing details for both streams Source/Sink.

### CODE REVIEW

File: docs/cep/sink/sink-types/stream-sink.md [hunk 1]
```diff
@@ -28,6 +40,8 @@ CREATE SINK STREAM SampleStreamSink (data string);
 |------------------|----------------------------------|---------------|---------------------|----------|
 | stream      | The streams to which the sink needs to publish events.|     | STRING        | No                  |
 | replication.type | Specifies if the replication type of the streams. Possible values can be `local` and `global`      | local         | STRING         | Yes      |
+| num.io.threads | The number of io threads | 1 | INT | Yes|
```
Comment (diana-macrometa) [inline docs/cep/sink/sink-types/stream-sink.md]:
```suggestion
| num.io.threads | The number of I/O threads. | 1 | INT | Yes|
```

Comment (diana-macrometa) [reply docs/cep/sink/sink-types/stream-sink.md]: What are "io" threads? Should it be IO, or perhaps I/O?

File: docs/cep/sink/sink-types/stream-sink.md [hunk 2]
```diff
@@ -28,6 +40,8 @@ CREATE SINK STREAM SampleStreamSink (data string);
...
+| key.shared.attributes | The attributes to be included into message's key| - | STRING | Yes|
```
Comment (diana-macrometa) [inline docs/cep/sink/sink-types/stream-sink.md:44]:
```suggestion
| key.shared.attributes | The attributes to be included into message's key. | - | STRING | Yes|
```

File: docs/cep/source/source-types/stream-source.md [hunk 3]
```diff
@@ -11,24 +11,50 @@ The stream processor groups common types of events together with a schema. This
...
+      receiver.queue.size="INT",      
+      map.type='type')  
+    (<attribute_name> <attribute_type>[,
+     <attribute_name> <attribute_type>] );
+```
+
+Or you can use the syntax shortcut `CREATE STREAM`:
+
+```sql
+CREATE STREAM <stream_name> <GLOBAL | LOCAL> (<attribute_name> <attribute_type>,
```
Comment (diana-macrometa) [inline docs/cep/source/source-types/stream-source.md:32]: Do the new parameters not work for CREATE STREAM?

File: docs/cep/source/source-types/stream-source.md [hunk 4]
```diff
@@ -11,24 +11,50 @@ The stream processor groups common types of events together with a schema. This
...
 
-| Parameter     | Description |
-| ------------- |-------------|
-| `stream name`      | The name of the stream created. (By convention, stream names use `PascalCase`.) |
-| `attribute name`   | Uniquely identifiable name of the stream attribute. (By convention, attribute names are defined in `camelCase`.)|     |
-| `attribute type`   | The type of each attribute defined in the schema. This can be `STRING`, `INT`, `LONG`, `DOUBLE`, `FLOAT`, `BOOL`, or `OBJECT`.     |
+| Parameter     | Description | Default Value | Possible Data Types | Optional |
+| ------------- |-------------| ------------- | ------------------- | -------- |
+| `stream.list` | The list of streams the sources will consume events.| -    | STRING        | No                  |
+| replication.type | Specifies if the replication type of the streams. Possible values can be `local` and `global`      | local         | STRING         | Yes      |
```
Comment (diana-macrometa) [inline docs/cep/source/source-types/stream-source.md]:
```suggestion
| replication.type | Specifies if the replication type of the streams. Possible values can be `local` and `global`.      | local         | STRING         | Yes      |
```

Comment (diana-macrometa) [reply docs/cep/source/source-types/stream-source.md]: Should we capitalize GLOBAL and LOCAL? We do elsewhere when talking about these parameters.

File: docs/cep/source/source-types/stream-source.md [hunk 5]
```diff
@@ -11,24 +11,50 @@ The stream processor groups common types of events together with a schema. This
...
+| `subscription.initial.position` | Subscription's initial position of the stream. Possible Values: [Earliest,Latest]| Lates | STRING | Yes|
```
Comment (diana-macrometa) [inline docs/cep/source/source-types/stream-source.md]:
```suggestion
| subscription.initial.position | Subscription's initial position in the stream. Possible values: [Earliest,Latest]| Latest | STRING | Yes|
```

File: docs/cep/source/source-types/stream-source.md [hunk 6]
```diff
@@ -11,24 +11,50 @@ The stream processor groups common types of events together with a schema. This
...
 The following parameters are used to configure a stream definition.
 
-| Parameter     | Description |
-| ------------- |-------------|
-| `stream name`      | The name of the stream created. (By convention, stream names use `PascalCase`.) |
-| `attribute name`   | Uniquely identifiable name of the stream attribute. (By convention, attribute names are defined in `camelCase`.)|     |
-| `attribute type`   | The type of each attribute defined in the schema. This can be `STRING`, `INT`, `LONG`, `DOUBLE`, `FLOAT`, `BOOL`, or `OBJECT`.     |
+| Parameter     | Description | Default Value | Possible Data Types | Optional |
+| ------------- |-------------| ------------- | ------------------- | -------- |
+| `stream.list` | The list of streams the sources will consume events.| -    | STRING        | No                  |
```
Comment (diana-macrometa) [inline docs/cep/source/source-types/stream-source.md]:
```suggestion
| stream.list | The list of streams the sources will consume events.| -    | STRING        | No                  |
```

File: docs/cep/source/source-types/stream-source.md [hunk 7]
```diff
@@ -11,24 +11,50 @@ The stream processor groups common types of events together with a schema. This
...
+| replication.type | Specifies if the replication type of the streams. Possible values can be `local` and `global`      | local         | STRING         | Yes      |
+| `subscription.initial.position` | Subscription's initial position of the stream. Possible Values: [Earliest,Latest]| Lates | STRING | Yes|
+| `subscription.type` | Subscription's type of the stream. Possible Values: [Exclusive, Shared, Failover, Key_Shared]. | Shared | STRING | Yes|
```
Comment (diana-macrometa) [inline docs/cep/source/source-types/stream-source.md]:
```suggestion
| `subscription.type` | Stream subscription type. Possible Values: [Exclusive, Shared, Failover, Key_Shared]. | Shared | STRING | Yes|
```

**Review by diana-macrometa — CHANGES_REQUESTED** [review]

**Comment by diana-macrometa** [comment]
Great content, Stoyan!

File: docs/cep/source/source-types/stream-source.md (hunk 4 above)
Comment (StoyanTc) [reply docs/cep/source/source-types/stream-source.md]: Yes, let's do it.

File: docs/cep/source/source-types/stream-source.md (hunk 3 above)
Comment (StoyanTc) [reply docs/cep/source/source-types/stream-source.md:32]: There is not a way to define them with `CREATE STREAM` shorthand. So, this construct will rely on the default values.

File: docs/cep/sink/sink-types/stream-sink.md (hunk 1 above)
Comment (StoyanTc) [reply docs/cep/sink/sink-types/stream-sink.md]: Yes, both are good for me.

File: docs/cep/source/source-types/stream-source.md [hunk 8]
```diff
@@ -11,24 +11,50 @@ The stream processor groups common types of events together with a schema. This
...
 
-| Parameter     | Description |
-| ------------- |-------------|
-| `stream name`      | The name of the stream created. (By convention, stream names use `PascalCase`.) |
-| `attribute name`   | Uniquely identifiable name of the stream attribute. (By convention, attribute names are defined in `camelCase`.)|     |
-| `attribute type`   | The type of each attribute defined in the schema. This can be `STRING`, `INT`, `LONG`, `DOUBLE`, `FLOAT`, `BOOL`, or `OBJECT`.     |
+| Parameter     | Description | Default Value | Possible Data Types | Optional |
+| ------------- |-------------| ------------- | ------------------- | -------- |
+| stream.list | The list of streams the sources will consume events.| -    | STRING        | No                  |
+| replication.type | Specifies if the replication type of the streams. Possible values can be `local` and `global`.      | local         | STRING         | Yes      |
```
Comment (diana-macrometa) [inline docs/cep/source/source-types/stream-source.md]:
```suggestion
| replication.type | Specifies if the replication type of the streams. Possible values can be `LOCAL` and `GLOBAL`.      | LOCAL         | STRING         | Yes      |
```

Comment (StoyanTc) [reply docs/cep/source/source-types/stream-source.md]: @diana-macrometa - Agreed.

File: docs/cep/source/source-types/stream-source.md [hunk 9]
```diff
@@ -11,24 +11,50 @@ The stream processor groups common types of events together with a schema. This
...
-| ------------- |-------------|
-| `stream name`      | The name of the stream created. (By convention, stream names use `PascalCase`.) |
-| `attribute name`   | Uniquely identifiable name of the stream attribute. (By convention, attribute names are defined in `camelCase`.)|     |
-| `attribute type`   | The type of each attribute defined in the schema. This can be `STRING`, `INT`, `LONG`, `DOUBLE`, `FLOAT`, `BOOL`, or `OBJECT`.     |
+| Parameter     | Description | Default Value | Possible Data Types | Optional |
+| ------------- |-------------| ------------- | ------------------- | -------- |
+| stream.list | The list of streams the sources will consume events.| -    | STRING        | No                  |
+| replication.type | Specifies if the replication type of the streams. Possible values can be `LOCAL` and `GLOBAL`.      | LOCAL         | STRING         | Yes      |
+| subscription.initial.position | Subscription's initial position in the stream. Possible values: [Earliest,Latest]| Latest | STRING | Yes|
+| `subscription.type` | Stream subscription type. Possible Values: [Exclusive, Shared, Failover, Key_Shared]. | Shared | STRING | Yes|
```
Comment (diana-macrometa) [inline docs/cep/source/source-types/stream-source.md]:
```suggestion
| subscription.type | Stream subscription type. Possible Values: [Exclusive, Shared, Failover, Key_Shared]. | Shared | STRING | Yes|
```

File: docs/cep/source/source-types/stream-source.md [hunk 10]
```diff
@@ -11,24 +11,50 @@ The stream processor groups common types of events together with a schema. This
...
+| `num.consumers` | Number of consumers | 1 | INT | Yes |
```
Comment (diana-macrometa) [inline docs/cep/source/source-types/stream-source.md]:
```suggestion
| num.consumers | Number of consumers. | 1 | INT | Yes |
```

File: docs/cep/source/source-types/stream-source.md [hunk 11]
```diff
@@ -11,24 +11,50 @@ The stream processor groups common types of events together with a schema. This
...
+| `num.listener.threads` | the number of listener threads | 1 | INT | Yes |
```
Comment (diana-macrometa) [inline docs/cep/source/source-types/stream-source.md]:
```suggestion
| num.listener.threads | The number of listener threads. | 1 | INT | Yes |
```

File: docs/cep/source/source-types/stream-source.md [hunk 12]
```diff
@@ -11,24 +11,50 @@ The stream processor groups common types of events together with a schema. This
...
+| `num.io.threads` | The number of listener threads | 1 | INT | Yes |
```
Comment (diana-macrometa) [inline docs/cep/source/source-types/stream-source.md]:
```suggestion
| num.io.threads | The number of listener threads. | 1 | INT | Yes |
```

File: docs/cep/source/source-types/stream-source.md (hunk 11 above)
Comment (diana-macrometa) [reply docs/cep/source/source-types/stream-source.md]: What do you mean by "threads" here? I have not seen "threads" used in our docs thus far. Do you mean streams? Subscriptions?

File: docs/cep/source/source-types/stream-source.md [hunk 13]
```diff
@@ -11,24 +11,50 @@ The stream processor groups common types of events together with a schema. This
...
+| `receiver.queue.size` | The number of messages accumulated by a consumer before an application calls | 1000 | INT | Yes |
```
Comment (diana-macrometa) [inline docs/cep/source/source-types/stream-source.md]:
```suggestion
| receiver.queue.size | The number of messages accumulated by a consumer before an application calls. | 1000 | INT | Yes |
```

**Review by diana-macrometa — CHANGES_REQUESTED** [review]

**Comment by diana-macrometa** [comment]
@mikeridley and @elof, FYI.

File: docs/cep/source/source-types/stream-source.md (hunk 11 above)
Comment (StoyanTc) [reply docs/cep/source/source-types/stream-source.md]: @diana-macrometa - The `threads` either `listener` or `io` are part of pulsar consumer spec. 
This is similar to the `threading.option` in `Kafka-source`.

Comment (diana-macrometa) [reply docs/cep/source/source-types/stream-source.md]: @mikeridley and @dustinlarimer, does this align with our product language for Macrometa?

**Comment by diana-macrometa** [comment]
@brandonLyonMM, this build is failing.

**Comment by brandonLyonMM** [comment]
> @brandonLyonMM, this build is failing.

@diana-macrometa this is the error logged:

```
[ERROR] Error: Docusaurus found broken links!
- On source page path = /docs/release-notes/release-notes-0-17-12:
--
09:47:50.415 | -> linking to ../cep/query-guide/functions/edge-functions/edge-functions (resolved as: /docs/cep/query-guide/functions/edge-functions/edge-functions)
```

**Comment by brandonLyonMM** [comment]
> > @brandonLyonMM, this build is failing.
> 
> @diana-macrometa this is the error logged:
> 
> ```
> [ERROR] Error: Docusaurus found broken links!
> - On source page path = /docs/release-notes/release-notes-0-17-12:
> --
> 09:47:50.415 | -> linking to ../cep/query-guide/functions/edge-functions/edge-functions (resolved as: /docs/cep/query-guide/functions/edge-functions/edge-functions)
> ```

@diana-macrometa this has been resolved

**Review by diana-macrometa — APPROVED** [review]

**Pull request merged**